scripts/clean_obsolete_strings.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python3
+
+'''
+This script is used to remove obsolete strings from all locales,
+and reformat files accordingly to the source content (en-US).
+'''
+
+from compare_locales.parser import getParser
+from compare_locales.serializer import serialize
+import os
+import subprocess
+import sys
+
+
+def extractFileList(repository_path):
+    '''
+        Extract the list of supported files. Store the relative path and ignore
+        specific folders for products we don't want to modify (e.g. SeaMonkey).
+    '''
+
+    supported_formats = [
+        '.dtd',
+        '.properties',
+        '.ftl',
+    ]
+
+    excluded_folders = (
+        'calendar',
+        'chat',
+        'editor',
+        'extensions',
+        'mail',
+        'other-licenses',
+        'suite',
+    )
+
+    file_list = []
+    for root, dirs, files in os.walk(repository_path, followlinks=True):
+        for filename in files:
+            filename = os.path.relpath(
+                os.path.join(root, filename),
+                repository_path
+            )
+            # Ignore specific products
+            if filename.startswith(excluded_folders):
+                continue
+            for supported_format in supported_formats:
+                if filename.endswith(supported_format):
+                    file_list.append(filename)
+    file_list.sort()
+
+    return file_list
+
+
+def main():
+    # Get absolute path of the repository's root from the script location
+    root_folder = os.path.abspath(os.path.join(
+        os.path.dirname(__file__), os.pardir))
+    config_file = os.path.join(root_folder, 'config', 'config')
+
+    if not os.path.exists(config_file):
+        print('ERROR: config file is missing')
+        sys.exit(1)
+
+    # Try importing the path to l10n clones from config/config
+    with open(config_file, 'r') as cfg:
+        lines = cfg.readlines()
+        for line in lines:
+            if line.startswith("l10n_clones_path"):
+                l10n_clones_path = line.split('=')[1].strip().strip('"')
+            if line.startswith("quarantine_path"):
+                quarantine_path = line.split('=')[1].strip().strip('"')
+
+    # Make sure paths are set correctly
+    try:
+        l10n_clones_path
+    except NameError:
+        print('l10n_clones_path is not defined in the config file')
+        sys.exit(1)
+    try:
+        quarantine_path
+    except NameError:
+        print('quarantine_path is not defined in the config file')
+        sys.exit(1)
```

Comment:
Now that you have two scripts that do this, maybe it's a good idea to factor this out?